### DISCUSSION TITLE
Run EdgeX 3.1 without consul

### DISCUSSION THREAD

**Discussion by SynchronicIT** [edgexfoundry/community#243] [Q&A]
Consul is currently used as Registry and Configuration Provider. We use EdgeX in small devices which are not always running. 
Consul is the only application / docker which doesn't like to get rebooted or being switched off for longer than a week. 

The actual added value vs. the cost in storage, ram and CPU is not in balance. 
Can we also run without Consul and how should that look like? 
And alternatively, could Consul be replaced by a much more lightweight application? (e.g. which uses the RedisDB only for some config and the rest is in-memory)

**Comment by cloudxxx8** [comment]
We plan to replace Consul with another self-developed lightweight Config Provider & Registry in v3.2.
For now, you can just disable the usage of Consul.

**Reply by cloudxxx8** [reply]
Set `EDGEX_CONFIG_PROVIDER` to none and `EDGEX_USE_REGISTRY` to false, and remove Consul from the docker-compose file
https://docs.edgexfoundry.org/3.1/microservices/configuration/CommonEnvironmentVariables/#edgex_config_provider
https://docs.edgexfoundry.org/3.1/microservices/configuration/CommonEnvironmentVariables/#edgex_use_registry

**Comment by qs-wang** [comment]
Which project is it? I am interesting the same to run the edgex with a lightweight "Consul". 

> We plan to replace Consul with another self-developed lightweight Config Provider & Registry in v3.2. For now, you can just disable the usage of Consul.

**Reply by cloudxxx8** [reply]
we develoloped our own service, which is contributed by Eaton and IOTech
https://github.com/edgexfoundry/edgex-go/pull/4790
